R/s3.R

```diff
@@ -74,15 +74,20 @@ parse_s3_path <- function(path) {
 #' @keywords Internal
 .url_parse_other <- function(url) {
   parsed_url <- httr::parse_url(url)
-  bucket <- gsub(x = parsed_url$path, pattern = "^/?([[a-z0-9\\.-]*)/.*", 
+  bucket <- gsub(x = parsed_url$path, pattern = "^/?([[a-z0-9\\:\\.-]*)/.*", 
                  replacement = "\\1", ignore.case = TRUE)
-  object <- gsub(x = parsed_url$path, pattern = "^/?([a-z0-9\\.-]*)/(.*)", 
+  object <- gsub(x = parsed_url$path, pattern = "^/?([a-z0-9\\:\\.-]*)/(.*)", 
```

Comment:
```suggestion
  object <- gsub(x = parsed_url$path, pattern = "^/?([a-z0-9:\\.-]*)/(.*)", 
```

`:` alone doesn't have a special meaning in regexes so I don't believe we need to escape it. Am I missing something?